src/components/widgets/pseudomutoProtocGenDoc.tsx

```
@@ -51,122 +53,130 @@ export const PseudomutoProtocGenDoc = ({
     })
   }, [apiBuild, selectedProgrammingLanguage])
   useEffect(() => {
-    const promise = scheme
-      ? fetch(scheme)
-          .then((res) => res.text())
-          .then((res) => yaml.load(res) as APISpec)
-      : undefined
-    fetch(file)
-      .then((res) => res.json())
-      .then((res) => {
-        const data = new APIDescriptionGeneratorModel()
-        data.setDescription(
-          'Documentation generated with github.com/gendocu-com/grpc-docs'
+    ;(async () => {
+      let apiSpec
+      if (scheme) {
+        const resp = await fetch(scheme)
+        apiSpec = yaml.load(await resp.text()) as APISpec
+      }
+      let protocDefinition
+      if (definition) {
+        protocDefinition = JSON.parse(definition)
+      } else if (file) {
+        const resp = await fetch(file)
+        protocDefinition = await resp.json()
+      }
+
+      if (!file && !definition) {
```

Can you also throw an error when both are defined?